pandas/core/indexes/multi.py

```diff
@@ -1244,7 +1244,9 @@ def _constructor(self) -> Callable[..., MultiIndex]:  # type: ignore[override]
         return type(self).from_tuples
 
     @doc(Index._shallow_copy)
-    def _shallow_copy(self, values: np.ndarray, name=lib.no_default) -> MultiIndex:
+    def _shallow_copy(
```

Comment:
if the options are doing this in take or shallow_copy, take is definitely preferred

Reply:
done -- implemented via take